packages/rpc-subscriptions-spec/src/rpc-subscriptions-channel.ts

```diff
@@ -26,3 +26,36 @@ export type RpcSubscriptionsChannelCreator<TOutboundMessage, TInboundMessage> =
         abortSignal: AbortSignal;
     }>,
 ) => Promise<RpcSubscriptionsChannel<TOutboundMessage, TInboundMessage>>;
+
+export function transformChannelInboundMessages<TOutboundMessage, TNewInboundMessage, TInboundMessage>(
+    channel: RpcSubscriptionsChannel<TOutboundMessage, TInboundMessage>,
+    transform: (message: TInboundMessage) => TNewInboundMessage,
+): RpcSubscriptionsChannel<TOutboundMessage, TNewInboundMessage> {
+    return Object.freeze<RpcSubscriptionsChannel<TOutboundMessage, TNewInboundMessage>>({
+        ...channel,
+        on(type, subscriber, options) {
+            if (type !== 'message') {
```

Comment:
You could take the channel name as an argument, and write some extra tests, but meh.

Reply:
Because the channel name can only be `"message"` or `"error"` for channels, I thought I'd focus on `"message"` here and, in the future, we can add a `transformChannelInboundErrors` helper if needed.